modules/apps/mail/mail-messaging-impl/src/main/java/com/liferay/mail/messaging/internal/MailServiceImpl.java

```diff
@@ -453,4 +471,78 @@ private Properties _getProperties(Account account) {
 		_mailSettingSystemConfiguration;
 	private final Map<Long, Session> _sessions = new ConcurrentHashMap<>();
 
+	private static class MailSettingCompanyConfigurationModelListener
+		implements ConfigurationModelListener {
+
+		public static void register(MailService mailService) {
+			BundleContext bundleContext = SystemBundleUtil.getBundleContext();
+
+			MailSettingCompanyConfigurationModelListener
+				mailSettingCompanyConfigurationModelListener =
+					new MailSettingCompanyConfigurationModelListener(
+						mailService);
+
+			mailSettingCompanyConfigurationModelListener._serviceRegistration =
+				bundleContext.registerService(
+					ConfigurationModelListener.class,
+					mailSettingCompanyConfigurationModelListener,
+					HashMapDictionaryBuilder.<String, Object>put(
+						"model.class.name", _PID
+					).build());
+
+			Map<String, Object> attributes =
+				LocalProcessLauncher.ProcessContext.getAttributes();
```

Comment:
@jeffwu0724 Could you explain a bit about why you save it into the `LocalProcessLauncher.ProcessContext`? Do we run this in a new process?